### PR TITLE
[frontend/backend] highlight selected filter

### PR DESCRIPTION
**Pull request by jpkha**
### Proposed changes

* Highlight the selected filters

### CODE REVIEW

**Comment by Archidoit** [comment]
The order of the values in a filter autocomplete is random within the list or withing a subgroup : 
image

image

Should we display the values in the alphabetical order?

File: opencti-platform/opencti-front/src/components/filters/FiltersModel.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface FilterRepresentatives {
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/components/filters/FiltersModel.ts]: rename in FilterRepresentative (with no final 'S')

**Comment by Archidoit** [comment]
Maybe for the 'label' filter we can put the 'no label' option at the top 
image

**Comment by Archidoit** [comment]
All the selected marking definitions are displayed with a green icon :
image

**Comment by Archidoit** [comment]
Problems with icon display for selected entities (ex here for a vocabulary) : 
image

File: opencti-platform/opencti-front/src/components/filters/FilterChipPopover.tsx [hunk 2]
```diff
@@ -223,25 +226,54 @@ export const FilterChipPopover: FunctionComponent<FilterChipMenuProps> = ({
...
+      <Autocomplete
+        multiple
         key={fKey}
-        selectOnFocus={true}
-        autoSelect={false}
-        autoHighlight={true}
         getOptionLabel={(option) => option.label ?? ''}
         noOptionsText={t_i18n('No available options')}
-        options={getOptionsFromEntities(entities, searchScope, fKey)}
-        groupBy={
```
Comment (jpkha) [inline opencti-platform/opencti-front/src/components/filters/FilterChipPopover.tsx:238]: Remove into a function but remove also the checking if isStixObjecTypes. In the searchEntitesUtil, we do a mapping where if `group `is undefined we use the `type `value.

File: opencti-platform/opencti-graphql/src/domain/stixCoreObject.js [hunk 3]
```diff
@@ -537,7 +537,12 @@ export const findFiltersRepresentatives = async (context, user, inputFilters) =>
       entity = await findStatusById(context, user, entity.id);
     }
     // add the entity representative in 'value', or null for deleted/restricted entities
-    filtersRepresentatives.push({ id: idsToResolve[index], value: (entity ? extractEntityRepresentativeName(entity) : null) });
+    filtersRepresentatives.push({
+      id: idsToResolve[index],
+      value: (entity ? extractEntityRepresentativeName(entity) : null),
+      entity_type: entity?.entity_type ?? null,
+      color: entity?.color || entity?.x_opencti_color || null
```
Comment (jpkha) [inline opencti-platform/opencti-graphql/src/domain/stixCoreObject.js:544]: I add your suggestion @Archidoit  👍

File: opencti-platform/opencti-front/src/utils/filters/SearchEntitiesUtil.ts [hunk 4]
```diff
@@ -18,15 +18,24 @@ export const getOptionsFromEntities = (
...
+        .filter((n) => searchScope[filterKey].some((s) => (n.parentTypes ?? []).concat(n.type).includes(s)));
     } else {
-      filteredOptions = (entities[filterKey] || []).sort((a, b) => (b.type ? -b.type.localeCompare(a.type) : 0));
+      filteredOptions = (entities[filterKey] || []);
     }
   } else if (entities[filterKey]) {
     filteredOptions = entities[filterKey];
   }
-  return filteredOptions;
+  return filteredOptions.map((f) => (f.group ? f : { ...f, group: f.type }))
```
Comment (jpkha) [inline opencti-platform/opencti-front/src/utils/filters/SearchEntitiesUtil.ts]: Here we map and sort every list.
So we do a mapping. Replace for the isStixObject case where `group ` is `undefined`.
Then we sort by group and in alphabetic order

**Comment by Archidoit** [comment]
When a deleted/no longer accessible entity is selected, it is displayed as 'Unknown' instead of 'deleted' : 
image

**Comment by Archidoit** [comment]
'selected entities' at the top is not working for 'in regards of' filter: 
image

**Comment by Archidoit** [comment]
Maybe write 'selected' with a first capitale letter ('Selected') 
image

**Review by Archidoit — APPROVED** [review]

**Pull request merged**